internal/core/parser.py

```diff
@@ -0,0 +1,60 @@
+"""
+The Parser class is used for argument parser operations.
+"""
+import argparse
+from internal.modules.auth.query import account_type
+from internal.core.keys import keys_show
+from internal.utils import validate_num_txs
+
+
+class ParseTestsDefaultFlags:  # pylint: disable=R0903
+    """
+    The ParseTestsDefaultFlags class is used for argument parser operations.
+    """
+
+    def __init__(  # pylint: disable=R0913
+        self,
+        desc,
+        sender=False,
+        sender_account="account1",
+        receiver=False,
+        num_txs=False,
+    ):
+        self.parser = argparse.ArgumentParser(desc)
+        self.sender = sender
+        self.receiver = receiver
+        self.num_txs = num_txs
+        if self.sender:
+            self.parser.add_argument(
+                "-s",
+                "--sender",
+                type=account_type,
+                default=keys_show(sender_account)[1]["address"],
+                help="Sender bech32 address",
+            )
+        if self.receiver:
+            self.parser.add_argument(
+                "-r",
+                "--receiver",
+                type=account_type,
+                default=keys_show("account2")[1]["address"],
+                help="Receiver bech32 address",
+            )
+        if self.num_txs:
+            self.parser.add_argument(
+                "-n",
+                "--num_txs",
+                type=validate_num_txs,
+                default=10000,
+                help="Number of transactions to be made, should be positive integer",
+            )
+
+    def get_args(self):
+        """
+        Returns the parsed arguments.
+        """
+        args = self.parser.parse_args()
+        sender = args.sender if self.sender else None
+        receiver = args.receiver if self.receiver else None
+        num_txs = int(args.num_txs) if self.num_txs else None
+        return sender, receiver, num_txs
```

Comment:
As per the name, it should return args (array). Not individual fields. We can just use `args.sender`, `args.receiver` etc when required
```suggestion
        args = self.parser.parse_args()
        args.sender = args.sender if self.sender else None
        args.receiver = args.receiver if self.receiver else None
        args.num_txs = int(args.num_txs) if self.num_txs else None
        return args
```

Do we really need to assign `None` here?

Reply:
We can return the array here, But yes we need None because we are not using the same args everywhere. some test scripts don't need `receiver` and `num_txs` args.

Reply:
And also anil. if we return array. we need to add extra code for accessing the values. We are currently following the below code.
```
def get_args(self):
    return sender, receiver, num_txs     
```
`sender, receiver, num_txs = p.get_args()`

Reply:
It quite makes it simple to access the values.

Reply:
But its a general definition. If other apps need to use it for different env settings, they would need to write another function. If we return all the args here, they can just call `args.sender`, `args.receiver`, `args.node` etc.

Reply:
merging this for now. Let's have another PR for that